app/models/comment.rb

```diff
@@ -64,6 +75,12 @@ def initialize(person, target, text)
       super(person, target)
     end
 
+    def build(*args)
+      super.tap do |comment|
+        comment.filter_mentions(comment.people_allowed_to_be_mentioned)
```

Comment:
I discovered that this incoming filter is a bit more complex than I thought first.

For **public** stuff it's easy: don't filter anything, the sender can mention anyone. You can't know the senders aspects, so you don't know if he is allowed to mention someone.

For **private comments** it's easy too: you can only mention other participants, every pod can validate and filter that.

For **private posts** it's a bit more complex: We can not know if the mentioned person can see the post, when we receive it (that's what I not considered before). So we can not filter anything here and need to trust the sender. But my notification logic already handles the notifications correctly, so the person only receives a notification, when he receives the visibility. So maybe we need to ensure that the person can not see the post on his stream, unless he also receives the visibility, instead of filter the mention?

So what you built now is, that you filter all mentions on private posts on receive, which makes mention impossible on private posts.